File: selfstudy/golang/stars_reverse.go
/*
task 288:
Stars - Reverse
You need to reverse the order of star luminosities
*/

package main

import (
	"fmt"
	"reflect"
)

func Reverse_orderStars(brightness []int) []int {
	res := make([]int, len(brightness))
	for i, v := range brightness {
		res[len(brightness)-1-i] = v
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Reverse_orderStars([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Reverse_orderStars([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Reverse_orderStars([]int{1, 2, 3})},
		{[]int{100, 200}, Reverse_orderStars([]int{100, 200})},
		{[]int{}, Reverse_orderStars([]int{})},
	}
	
	for idx, t := range tests {
		res := Reverse_orderStars(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/stars_reverse.js
/*
task 88:
Stars - Reverse
You need to reverse the order of star luminosities
*/

function reverse_order_stars(brightness) {
  return [...brightness].reverse();
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: reverse_order_stars([10, 20, 30]) },
  { input: [50, 60, 70], expected: reverse_order_stars([50, 60, 70]) },
  { input: [1, 2, 3], expected: reverse_order_stars([1, 2, 3]) },
  { input: [100, 200], expected: reverse_order_stars([100, 200]) },
  { input: [], expected: reverse_order_stars([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('reverse_order_stars', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(reverse_order_stars(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, reverse_order_stars(t.input) === t.expected || JSON.stringify(reverse_order_stars(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works
